### PR TITLE
[SmartCamera]Agent mode

### PR DESCRIPTION
**Pull request by ayinloya**
https://app.shortcut.com/smileid/story/12882/add-agent-mode-support-to-the-web-components

This adds agent mode to web components
When the `allow_agent_mode` flag is set to true. The button to switch between the agent mode and selfie mode will be visible.

# Testing
- Run the example app with the allow agent mode flag and verify that the button exists.
- Test on a mobile device and confirm switching works

### CODE REVIEW

File: packages/web-components/components/selfie/src/selfie-capture/SelfieCapture.js [hunk 1]
```diff
@@ -557,12 +561,16 @@ class SelfieCaptureScreen extends HTMLElement {
...
+    if (this.allowAgentMode) {
+      this.facingMode = 'environment';
+    }
   }
 
   connectedCallback() {
     const template = document.createElement('template');
     template.innerHTML = this.render();
-
+    this.shadowRoot.innerHTML = '';
```
Comment (beaglebets) [inline packages/web-components/components/selfie/src/selfie-capture/SelfieCapture.js:573]: why did we need to do this?

File: packages/web-components/components/selfie/src/selfie-capture/SelfieCapture.js [hunk 2]
```diff
@@ -816,9 +848,52 @@ class SelfieCaptureScreen extends HTMLElement {
...
+          device.label.toLowerCase().includes('rear')
+        ) {
+          hasBackCamera = true;
+          return true;
+        }
+        return false;
+      });
+
+      if (hasBackCamera) {
+        console.warn('Device has a back camera.');
```
Comment (beaglebets) [inline packages/web-components/components/selfie/src/selfie-capture/SelfieCapture.js:887]: i think that a) we shouldn't be console.warning something that is a successful operation and b) we should throw and handle (with a user facing message) when someone tries to use agent mode on a device which doesn't support it.

File: packages/web-components/cypress/pages/smart-camera-web-agent-mode.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    <smart-camera-web
+      disable-image-tests
+      allow-agent-mode="true"
+    ></smart-camera-web>
+
+    <script src="./js/components/smart-camera-web/src/SmartCameraWeb.js"></script>
+    <script>
+      const app = document.querySelector('smart-camera-web');
+      app.addEventListener('smart-camera-web.published', async (e) => {
+        console.log(e.detail);
```
Comment (beaglebets) [inline packages/web-components/cypress/pages/smart-camera-web-agent-mode.html:37]: i don't think we want a console.log here. why is this here?

**Review by beaglebets — CHANGES_REQUESTED** [review]

File: packages/web-components/components/selfie/src/selfie-capture/SelfieCapture.js (hunk 1 above)
Comment (ayinloya) [reply packages/web-components/components/selfie/src/selfie-capture/SelfieCapture.js:573]: This is so that the view will be reset when the component is rerendered

File: packages/web-components/components/selfie/src/selfie-capture/SelfieCapture.js (hunk 2 above)
Comment (ayinloya) [reply packages/web-components/components/selfie/src/selfie-capture/SelfieCapture.js:887]: removed

File: example/script.js [hunk 4]
```diff
@@ -72,10 +72,11 @@ export default function setupForm() {
           product,
           callback_url,
           environment,
-          // demo_mode: true,
+          // // demo_mode: true,
```
[the remark this answers is not in the document]
Comment (ayinloya) [reply example/script.js]: fixed, left the commented out code there to easily now what configs can be added

**Review by beaglebets — APPROVED** [review]
